### ISSUE TITLE
BadImageFormatException

### ISSUE THREAD

**Issue by MutterOberin** [accord-net/framework#311]
I get it when trying to access Accord.Video.FFMPEG:

Additional information: Could not load file or assembly 'Accord.Video.FFMPEG, Version=3.3.0.0, Culture=neutral, PublicKeyToken=fa1a88e29555ccf7' or one of its dependencies. An attempt was made to load a program with an incorrect format.

Do I miss some dependencies?

**Comment by cesarsouza** [comment]
Thanks for opening the issue!

A likely cause for this exception is that you are attempting to compile and run your program in x86_64 (64-bits). Right now, Accord.Video.FFMPEG can only be used from 32-bit applications. If the issue persists, please let me know!  

Regards,
Cesar

**Comment by MutterOberin** [comment]
Thanks for the quick answer! 

I will give that a try! Are there any plans for 64-bit version? It is limited by the FFMPEG, not by Accord itself, right?

**Comment by cesarsouza** [comment]
Unfortunately it is a bit more complicated than that. The limitation is actually from the Accord side. The main problem is that the FFMPEG assembly has been created a few years ago when the FFMPEG API was completely different that what is now. At some point during development history (Video.FFMPEG was a library from AForge.NET at the time), there was a very large, compatibility breaking API change done by the FFMPEG authors. The problem is that in order to update the project to use a more recent and 64-bits compatible version of FFMPEG it would simple require Accord.Video.FFMPEG to be rewritten from scratch.

I have tried to perform this change about two years ago, without success (mostly due to lack or conflicting FFMPEG documentation after the change). But I can give it a try again, to see if things have improved. If anyone else would like to give it a try, please feel free as well.

**Comment by MutterOberin** [comment]
Thanks for the information!

First of all: yes it works in 32 bit. Since this limits the whole application in usable memory it would be much better to have a 64 bit version. I used Aforge FFMPEG before and due to the lack of an recent version I switched to Accord. When I remember correctly Aforge supported 64 bit. On the other hand I was unable to get it to run with .NET 4.6 and VS 2015. 

If I find some time I would support you in rewriting the Accord.Video.FFMPEG part to be used with recent FFMPEG versions. But I cant promise anything at this moment.
